source/main/gfx/GfxActor.cpp

```diff
@@ -114,6 +114,9 @@ RoR::GfxActor::GfxActor(Actor* actor, std::string ogre_resource_group,
 
 RoR::GfxActor::~GfxActor()
 {
+    this->FinishWheelUpdates();
+    this->FinishFlexbodyTasks();
```

Comment:
TDev-style workarounds ☹️ We have to schedule the disposal the same way we schedule spawn.

Reply:
Like this?
```
for (auto t: m_flexwheel_tasks) { t->join(); }
for (auto t: m_flexbody_tasks) { t->join(); }
```

Reply:
No, I mean the tasks should be started and stopped [only in a single location](https://github.com/RigsOfRods/rigs-of-rods/blob/742bff1ce076d02fa617b1be274ff6c20d6f699a/source/main/gfx/DustManager.cpp#L228-L233) and GfxActor should be destroyed while they're _definitely_ not running. Otherwise the design is flawed.

Reply:
I think I failed. I cannot reproduce the segfault anymore on latest master and I also don't understand how it could segfault there in the first place. I will revert the change.